test/Divvi.ts

```diff
@@ -148,4 +148,163 @@ describe(CONTRACT_NAME, () => {
       expect(await contract.defaultAdmin()).to.equal(otherAddress)
     })
   })
+
+  describe('transfer permission restrictions', () => {
+    it('should prevent non-permitted users from transferring', async function () {
+      const { contract, otherAccount, owner } = await deployDivviContract()
+
+      const contractAsNonPermittedAccount = contract.connect(
+        otherAccount,
+      ) as typeof contract
+
+      await expect(
+        contractAsNonPermittedAccount.transfer(mockAccount1, 500),
+      ).to.be.revertedWithCustomError(contract, 'SenderNotPermitted')
+      await expect(
+        contractAsNonPermittedAccount.transferFrom(mockAccount1, owner, 500),
+      ).to.be.revertedWithCustomError(contract, 'SenderNotPermitted')
+    })
+
+    it('should allow permitted users to transfer', async function () {
+      const { contract, minterAccount, otherAccount } =
+        await deployDivviContract()
+
+      await contract.setSenderPermission(otherAccount, true)
+
+      // Mint tokens to permitted user
+      const contractAsMinter = contract.connect(
+        minterAccount,
+      ) as typeof contract
+      await contractAsMinter.mint(otherAccount, 1000n)
+
+      const contractAsPermittedAccount = contract.connect(
+        otherAccount,
+      ) as typeof contract
+      // transfer all tokens from otherAccount to owner
+      await contractAsPermittedAccount.transfer(minterAccount.address, 1000n)
+
+      // Assert balances after transfer
+      expect(await contract.balanceOf(otherAccount.address)).to.equal(0n)
+      expect(await contract.balanceOf(minterAccount.address)).to.equal(1000n)
+    })
+
+    it('should allow permitted users to call transferFrom', async function () {
```

Comment:
nit: this test name is slightly misleading, should it be something like "should allow spender to transferFrom permitted users"? since the address _calling_ `transferFrom` doesn't actually need to be permitted.